alchemy/src/cloudflare/api.ts

```diff
@@ -42,56 +51,109 @@ export interface CloudflareApiOptions {
   email?: string;
 }
 
-/** Used to propagate normalized auth options from a parent resource to `createCloudflareApi` in a child resource */
-export type InternalCloudflareApiOptions = CloudflareAuthOptions & {
-  baseUrl?: string;
-  accountId: string;
-};
-
-function computeCacheKey(options: CloudflareApiOptions): string {
-  return `${options.baseUrl}|${options.accountId}|${options.apiKey?.unencrypted}|${options.apiToken?.unencrypted}|${options.email}`;
-}
-
-const cloudflareApiCache: Record<string, CloudflareApi> = {};
-
 /**
  * Creates a CloudflareApi instance with automatic account ID discovery if not provided
  *
  * @param options API options
  * @returns Promise resolving to a CloudflareApi instance
  */
-export async function createCloudflareApi(
-  options: Partial<CloudflareApiOptions> | InternalCloudflareApiOptions = {},
-): Promise<CloudflareApi> {
-  // TODO: Implement scope-level credential resolution similar to AWS
-  // This function should check for scope.providerCredentials.cloudflare
-  // and merge those credentials with the provided options, following
-  // the same three-tier resolution pattern: global → scope → resource
-  const cacheKey = computeCacheKey(options);
-  if (cloudflareApiCache[cacheKey]) {
-    return cloudflareApiCache[cacheKey];
-  }
+export const createCloudflareApi = memoize(
+  async (options: CloudflareApiOptions = {}) => {
+    const baseUrl = options.baseUrl ?? process.env.CLOUDFLARE_BASE_URL;
+    const apiKey =
+      options.apiKey?.unencrypted ?? process.env.CLOUDFLARE_API_KEY;
+    const apiToken =
+      options.apiToken?.unencrypted ?? process.env.CLOUDFLARE_API_TOKEN;
+    const email = options.email ?? process.env.CLOUDFLARE_EMAIL;
+    const accountId =
+      options.accountId ??
+      process.env.CLOUDFLARE_ACCOUNT_ID ??
+      process.env.CF_ACCOUNT_ID;
 
-  const authOptions = await normalizeAuthOptions(options);
-  const accountId =
-    options.accountId ??
-    process.env.CLOUDFLARE_ACCOUNT_ID ??
-    process.env.CF_ACCOUNT_ID ??
-    (await getCloudflareAccountId(authOptions));
-  return (cloudflareApiCache[cacheKey] = new CloudflareApi({
-    baseUrl: options.baseUrl,
-    accountId,
-    authOptions,
-  }));
-}
+    if (options.profile) {
```

Comment:
I think we mentioned `API Key > API Token > Explicit Profile > Default Profile`

Reply:
Made that update but I feel like `options.profile` should probably take precedence over an API key or token in your environment? I don't have too strong an option on this though.